### ISSUE TITLE
Help modal dialog: inaccurate description for the Insert link keyboard shortcut

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#54148]
### Description

Noticed while looking into https://github.com/WordPress/gutenberg/issues/51737

In the Keyboard shortcuts modal dialog, the description for the Ctrl/Cmd+k shortcut is:

`Convert the selected text into a link.`

That's inaccurate. The shortcut works also without a selection, when the cursor is within any block with text. It works also when the cursor isn't inside any text and it's actially in an empty space. That's how it always worked in WordPress since ages. There are good reasons for this behavior, as it provides users with more flexibility.

However, the current description actually hides how this feature works and it's misleading. It doesn't help with discoverability. The descritpion should be simpler and mroe generic, to match the actual behavior:
- No text selection is required to add a link.
- Links can be added also to other elements, for example an inline image within a paragraph block. The description only mentions text so it's inaccurate and pointlessy too specific.

### Step-by-step reproduction instructions

- Go to the Post Editor > Options > Keyboard shortcuts
- The Keyboard shortcuts modal dialog opens.
- Observe the description of the Ctrl/Cmd+k keyboard shortcut within the section 'Text formatting'.

### Screenshots, screen recording, code snippet

<img width="616" alt="Screenshot 2023-09-04 at 10 11 53" src="https://github.com/WordPress/gutenberg/assets/1682452/3f26b226-4d60-4236-98d2-d0922bc9c83c">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
For reference, the Classic Editor explains it as `Insert/edit link (Ctrl+K)`. How about `Insert or edit a link.` etc.?

![classic-editor](https://github.com/WordPress/gutenberg/assets/54422211/797095f1-2a1f-4f90-9d0f-54508814365b)

**Comment by karthick-murugan** [comment]
@afercia - I just tested this issue in the latest gutenberg version 19.9.0 and it seems if CMD K is pressed without selection the command palette is opened and CMD K is pressed with selection, the link option is opened. Also tested for image block, with selection and only the command palette is opened and not the link option. Here is the tested video for your reference. 

https://github.com/user-attachments/assets/948f0b02-30c2-4c3b-867a-e1c17dc5fbd6

**Comment by t-hamano** [comment]
> it seems if CMD K is pressed without selection the command palette is opened and CMD K is pressed with selection, the link option is opened.

That's right. https://github.com/WordPress/gutenberg/pull/66056 changed it to work this way.

**Comment by karthick-murugan** [comment]
Thanks for the confirmation @t-hamano. Hope @afercia can check and close this issue.

**Comment by afercia** [comment]
> That's right. [#66056](https://github.com/WordPress/gutenberg/pull/66056) changed it to work this way.

In my opinion, https://github.com/WordPress/gutenberg/pull/66056 should be reverted as it changes a long established interaction pattern. It cures the symptom and not the cause. I will comment on that PR and create a new issue.
